ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -373,6 +382,20 @@ public Query visitQuery(final SqlBaseParser.QueryContext context) {
           visit(context.selectItem(), SelectItem.class)
       );
 
+      final boolean staticQuery = context.WITH() != null
+          || (context.EMIT() == null && !buildingPersistentQuery);
+
+      final ResultMaterialization resultMaterialization = Optional
+          .ofNullable(context.resultMaterialization())
```

Comment:
can I pretty please ask you to make it the normal `if-else` style of coding? I feel like this is abusing Optional haha 😂

Reply:
Nope, sorry.